src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -74,6 +74,16 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
         /// </summary>
         private readonly Dictionary<VisualStudioMetadataReference, CancellationTokenSource> _donotAccessDirectlyChangedReferencesPendingUpdate
             = new Dictionary<VisualStudioMetadataReference, CancellationTokenSource>();
+
+        /// <summary>
+        /// Holds the "original" snapshot in the project that needs to be updated due a reference changing. Because we defer updating after a metadata
+        /// reference change, we need to rememeber the original before so we can do the removal/add correctly.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="OnImportChangedAfterDelay(System.Threading.Tasks.Task, object)" /> is what consumes this.</remarks>
+        private readonly Dictionary<VisualStudioMetadataReference, PortableExecutableReference> _originalSnapshotOfReferenceNeedingUpdate
```

Comment:
>_originalSnapshotOfReferenceNeedingUpdate [](start = 96, length = 41)

It's not clear that this just adds another piece of information to the references queued in ChangedReferencesPendingUpdate. Perhaps merge these two dictionaries or add a comment.

Reply:
@tmat Merged the dictionaries now. Want to re-review?